### ISSUE TITLE
adminEnroll.html throws HTTP 500

### ISSUE THREAD

**Issue by SilleBille** [SilleBille/pki#508]
This issue was migrated from [Pagure Issue #3111](https://pagure.io/dogtagpki/issue/3111).Originally filed by **dglauche** on *2019-10-29*

 I tried the CA Admin Certificate Recovery process as described [here](https://www.dogtagpki.org/wiki/CA_Admin_Setup)

Unfortunately visiting the admin enroll page yields a HTTP 500 Internal Server Error.
Thats everything I got from the logs:
```
29-Oct-2019 16:44:56.244 SCHWERWIEGEND [https-jsse-nio-8443-exec-1] org.apache.catalina.core.StandardHostValve.invoke Exception Processing /ca/admin/ca/adminEnroll.html
	javax.ws.rs.ServiceUnavailableException: Subsystem unavailable
		at com.netscape.cms.tomcat.ProxyRealm.findSecurityConstraints(ProxyRealm.java:150)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:515)
		at com.netscape.cms.tomcat.ExternalAuthenticationValve.invoke(ExternalAuthenticationValve.java:82)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1589)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Thread.java:748)
```

The pki-server command states that the Instance is up and running:
```
pki-server status sample_ca
  Instance ID: sample_ca
  Active: True
  Unsecure Port: 8080
  Secure Port: 8443
  Tomcat Port: 8005

  CA Subsystem:
    Type:                Root CA (Security Domain)
    SD Registration URL: https://XXXX:8443
    Enabled:             True
    Unsecure URL:        http://XXXX:8080/ca/ee/ca
    Secure Agent URL:    https://XXXX:8443/ca/agent/ca
    Secure EE URL:       https://XXXX:8443/ca/ee/ca
    Secure Admin URL:    https://XXXX8443/ca/services
    PKI Console URL:     https://XXXX:8443/ca
```

I tried to get some debugging information by setting the debug.level to 5 as mentioned in the [wiki](https://www.dogtagpki.org/wiki/PKI_10.7_Server_Debug_Log) but the server refuses the write debug logs.

Um running Fedora 30 with dogtag-pki 10.7.3

**Comment by SilleBille** [comment]
Posted by cipherboy on 2019-10-29: 

One thing you might trying is checking the `journalctl -u pki-tomcatdsample_ca.service` output for your tomcat instance -- when debugs aren't getting written, typically it means not enough of Dogtag has been started to write logs yet. So check `journalctl` for that output.

**Comment by SilleBille** [comment]
Posted by dglauche on 2019-10-30: 

Using debug level 5 I was able to get these logs:
```
...
2019-10-30 14:31:21 [main] FINE: GenericPolicyProcessor::init Certificate Policy Framework (deprecated) is ENABLED
2019-10-30 14:31:21 [main] SEVERE: No policy implementation exists for: null: com.netscape.cms.policy.constraints.ManualAuthentication
java.lang.ClassNotFoundException: com.netscape.cms.policy.constraints.ManualAuthentication
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1365)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1188)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at org.dogtagpki.legacy.core.policy.GenericPolicyProcessor.initSystemPolicies(GenericPolicyProcessor.java:1220)
	at org.dogtagpki.legacy.core.policy.GenericPolicyProcessor.init(GenericPolicyProcessor.java:202)
	at org.dogtagpki.legacy.ca.CAPolicy.init(CAPolicy.java:84)
	at com.netscape.ca.CertificateAuthority.initRequestQueue(CertificateAuthority.java:2214)
	at com.netscape.ca.CertificateAuthority.init(CertificateAuthority.java:590)
	at com.netscape.cmscore.apps.CMSEngine.initSubsystem(CMSEngine.java:817)
	at com.netscape.cmscore.apps.CMSEngine.initSubsystems(CMSEngine.java:684)
	at com.netscape.cmscore.apps.CMSEngine.initSubsystems(CMSEngine.java:676)
	at com.netscape.cmscore.apps.CMSEngine.init(CMSEngine.java:457)
	at com.netscape.cms.servlet.base.CMSStartServlet.init(CMSStartServlet.java:105)
	at javax.servlet.GenericServlet.init(GenericServlet.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:282)
	at org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:279)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAsPrivileged(Subject.java:549)
	at org.apache.catalina.security.SecurityUtil.execute(SecurityUtil.java:314)
	at org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:170)
	at org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:123)
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1122)
	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1089)
	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:983)
	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:4871)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5180)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:717)
	at org.apache.catalina.core.ContainerBase.access$000(ContainerBase.java:129)
	at org.apache.catalina.core.ContainerBase$PrivilegedAddChild.run(ContainerBase.java:150)
	at org.apache.catalina.core.ContainerBase$PrivilegedAddChild.run(ContainerBase.java:140)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:688)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:631)
	at org.apache.catalina.startup.HostConfig$DeployDescriptor.run(HostConfig.java:1831)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:526)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:425)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1576)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:309)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:936)
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:841)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909)
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:421)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:932)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:633)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:344)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:475)

2019-10-30 14:31:21 [main] SEVERE: Unable to start CMS engine: No policy implementation exists for: null
No policy implementation exists for: null
	at org.dogtagpki.legacy.core.policy.GenericPolicyProcessor.initSystemPolicies(GenericPolicyProcessor.java:1248)
	at org.dogtagpki.legacy.core.policy.GenericPolicyProcessor.init(GenericPolicyProcessor.java:202)
	at org.dogtagpki.legacy.ca.CAPolicy.init(CAPolicy.java:84)
	at com.netscape.ca.CertificateAuthority.initRequestQueue(CertificateAuthority.java:2214)
	at com.netscape.ca.CertificateAuthority.init(CertificateAuthority.java:590)
	at com.netscape.cmscore.apps.CMSEngine.initSubsystem(CMSEngine.java:817)
	at com.netscape.cmscore.apps.CMSEngine.initSubsystems(CMSEngine.java:684)
	at com.netscape.cmscore.apps.CMSEngine.initSubsystems(CMSEngine.java:676)
	at com.netscape.cmscore.apps.CMSEngine.init(CMSEngine.java:457)
	at com.netscape.cms.servlet.base.CMSStartServlet.init(CMSStartServlet.java:105)
	at javax.servlet.GenericServlet.init(GenericServlet.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:282)
	at org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:279)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAsPrivileged(Subject.java:549)
	at org.apache.catalina.security.SecurityUtil.execute(SecurityUtil.java:314)
	at org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:170)
	at org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:123)
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1122)
	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1089)
	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:983)
	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:4871)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5180)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:717)
	at org.apache.catalina.core.ContainerBase.access$000(ContainerBase.java:129)
	at org.apache.catalina.core.ContainerBase$PrivilegedAddChild.run(ContainerBase.java:150)
	at org.apache.catalina.core.ContainerBase$PrivilegedAddChild.run(ContainerBase.java:140)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:688)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:631)
	at org.apache.catalina.startup.HostConfig$DeployDescriptor.run(HostConfig.java:1831)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:526)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:425)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1576)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:309)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:936)
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:841)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909)
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:421)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:932)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:633)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:344)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:475)

2019-10-30 14:31:21 [main] INFO: Shutting down.
...
```
```
...
2019-10-30 14:31:21 [main] FINE: CMSEngine: debug stopped
2019-10-30 14:31:21 [main] SEVERE: Servlet [castart] in web application [/ca] threw load() exception
No policy implementation exists for: null
	at org.dogtagpki.legacy.core.policy.GenericPolicyProcessor.initSystemPolicies(GenericPolicyProcessor.java:1248)
	at org.dogtagpki.legacy.core.policy.GenericPolicyProcessor.init(GenericPolicyProcessor.java:202)
	at org.dogtagpki.legacy.ca.CAPolicy.init(CAPolicy.java:84)
	at com.netscape.ca.CertificateAuthority.initRequestQueue(CertificateAuthority.java:2214)
	at com.netscape.ca.CertificateAuthority.init(CertificateAuthority.java:590)
	at com.netscape.cmscore.apps.CMSEngine.initSubsystem(CMSEngine.java:817)
	at com.netscape.cmscore.apps.CMSEngine.initSubsystems(CMSEngine.java:684)
	at com.netscape.cmscore.apps.CMSEngine.initSubsystems(CMSEngine.java:676)
	at com.netscape.cmscore.apps.CMSEngine.init(CMSEngine.java:457)
	at com.netscape.cms.servlet.base.CMSStartServlet.init(CMSStartServlet.java:105)
	at javax.servlet.GenericServlet.init(GenericServlet.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:282)
	at org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:279)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAsPrivileged(Subject.java:549)
	at org.apache.catalina.security.SecurityUtil.execute(SecurityUtil.java:314)
	at org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:170)
	at org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:123)
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1122)
	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1089)
	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:983)
	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:4871)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5180)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:717)
	at org.apache.catalina.core.ContainerBase.access$000(ContainerBase.java:129)
	at org.apache.catalina.core.ContainerBase$PrivilegedAddChild.run(ContainerBase.java:150)
	at org.apache.catalina.core.ContainerBase$PrivilegedAddChild.run(ContainerBase.java:140)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:688)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:631)
	at org.apache.catalina.startup.HostConfig$DeployDescriptor.run(HostConfig.java:1831)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
	at org.apache.catalina.startup.HostConfig.deployDescriptors(HostConfig.java:526)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:425)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1576)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:309)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:936)
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:841)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909)
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:421)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:932)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:633)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:344)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:475)
```

**Comment by SilleBille** [comment]
Posted by dmoluguw on 2019-10-30: 

the issue seems to be due to loading [1] a hard-coded class: `com.netscape.cms.policy.constraints.ManualAuthentication`



For v10.7.3, the `ManualAuthentication.java` should have been loaded as `org.dogtagpki.legacy.server.policy.constraints.ManualAuthentication` and the source exists in [2] 

The issue was introduced by commit [3]

[1] https://github.com/dogtagpki/pki/blob/master/base/server/src/org/dogtagpki/legacy/core/policy/GenericPolicyProcessor.java#L76
[2] https://github.com/dogtagpki/pki/blob/v10.7.3/base/server/cms/src/org/dogtagpki/legacy/server/policy/constraints/ManualAuthentication.java
[3] https://github.com/dogtagpki/pki/commit/54c04a1d7c270f4e162695105468d280bdc7d028#diff-a7f182e298c699927eedf50c57e010f9

**Comment by SilleBille** [comment]
Posted by edewata on 2019-11-04: 

This issue has been fixed upstream:

* master (10.8) branch: https://github.com/dogtagpki/pki/commit/46df45c0331312fb360bfebf3c38e003420596dc
* 10.7 branch: https://github.com/dogtagpki/pki/commit/b2a58eb240a93fe01e75311611341ed33f650619

**Comment by SilleBille** [comment]
Posted by dkt on 2019-11-05: 

Hello,

I am not sure if my problem is same as this bug. Originally the stack trace is the same as the original post. (I am using F31)

Then I had upgraded the jss in a recent update. Also I tried to get the pki-server srpm and update the GenericPolicyProcessor.java according to the git commit. Also recompiled the pki-* rpm and reinstall it.

Now the stack trace is different:

Nov 05 12:22:15 server[18504]: WARNING: Exception processing realm [com.netscape.cms.tomcat.ProxyRealm6f8609c8] background process
Nov 05 12:22:15 server[18504]: javax.ws.rs.ServiceUnavailableException: Subsystem unavailable
Nov 05 12:22:15 server[18504]:  at com.netscape.cms.tomcat.ProxyRealm.backgroundProcess(ProxyRealm.java:142)
Nov 05 12:22:15 server[18504]:  at org.apache.catalina.core.ContainerBase.backgroundProcess(ContainerBase.java:1137)
Nov 05 12:22:15 server[18504]:  at org.apache.catalina.core.StandardContext.backgroundProcess(StandardContext.java:5566)
Nov 05 12:22:15 server[18504]:  at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.processChildren(ContainerBase.java:1353)
Nov 05 12:22:15 server[18504]:  at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.processChildren(ContainerBase.java:1357)
Nov 05 12:22:15 server[18504]:  at org.apache.catalina.core.ContainerBase$ContainerBackgroundProcessor.run(ContainerBase.java:1335)
Nov 05 12:22:15 server[18504]:  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
Nov 05 12:22:15 server[18504]:  at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
Nov 05 12:22:15 server[18504]:  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
Nov 05 12:22:15 server[18504]:  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
Nov 05 12:22:15 server[18504]:  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Nov 05 12:22:15 server[18504]:  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Nov 05 12:22:15 server[18504]:  at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
Nov 05 12:22:15 server[18504]:  at java.lang.Thread.run(Thread.java:748)

Thanks.
